src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterClause.java

```diff
@@ -42,7 +42,11 @@ public void validateLogicalExpression(
   public void validateComparisonExpression(
       ComparisonExpression comparisonExpression, DocumentProjector indexingProjector) {
     String path = comparisonExpression.getPath();
-    boolean isPathIndexed = indexingProjector.isPathIncluded(path);
+    boolean isPathIndexed =
+        !indexingProjector.isIndexingDenyAll() && indexingProjector.isPathIncluded(path);
+
+    // special case path may be set to indexed for `$vector` and `$vectorize` fields even in case of
```

Comment:
Not 100% sure what this means? Does this relate to `isPathIndexed` determination above?

Reply:
Incase of deny all, indexing projection sets `$vector` and `$vectorize` as allowed fields. If we remove this those fields are getting removed while inserting data and not indexed, This is a hack to ignore them as indexed fields if indexing options is deny all.

Reply:
Ok so comment refers to line above? Makes sense.